File: test/language/choice_types/cpp98/FunctionSelectorChoiceTest.cpp
#include "gtest/gtest.h"

#include "choice_types/function_selector_choice/TestChoice.h"

// just testChoice setters and getters
namespace choice_types
{
namespace function_selector_choice
{

TEST(FunctionSelectorChoiceTest, field8)
{
    Selector selector;
    selector.setNumBits(8);
    TestChoice testChoice;
    testChoice.initialize(selector);
    testChoice.setField8(0x7F);
    ASSERT_EQ(0x7F, testChoice.getField8());
    ASSERT_EQ(8, testChoice.bitSizeOf());
}

TEST(FunctionSelectorChoiceTest, field16)
{
    Selector selector;
    selector.setNumBits(16);
    TestChoice testChoice;
    testChoice.initialize(selector);
    testChoice.setField16(0x7F7F);
    ASSERT_EQ(0x7F7F, testChoice.getField16());
    ASSERT_EQ(16, testChoice.bitSizeOf());
}

} // namespace function_selector_choice
} // namespace choice_types

File: test/language/choice_types/java/choice_types/FunctionSelectorChoiceTest.java
package choice_types;

import static org.junit.jupiter.api.Assertions.*;
import org.junit.jupiter.api.Test;

import choice_types.function_selector_choice.Selector;
import choice_types.function_selector_choice.TestChoice;

public class FunctionSelectorChoiceTest
{
    @Test
    public void field8()
    {
        final Selector selector = new Selector((byte)8);
        final TestChoice testChoice = new TestChoice(selector);
        testChoice.setField8((byte)0x7F);

        assertEquals(0x7F, testChoice.getField8());
        assertEquals(8, testChoice.bitSizeOf());
    }

    @Test
    public void field16()
    {
        final Selector selector = new Selector((byte)16);
        final TestChoice testChoice = new TestChoice(selector);
        testChoice.setField16((short)0x7F7F);

        assertEquals(0x7F7F, testChoice.getField16());
        assertEquals(16, testChoice.bitSizeOf());
    }
}
